components/TemporaryFlashcardsModal/TemporaryFlashcardsModal.js

```diff
@@ -0,0 +1,272 @@
+                          offColor="#e76f51"
+                          onColor="#2a9d8f"
+                          uncheckedIcon={false}
+                          checkedIcon={false}
+                          checked={temporaryFlashcard.isIncluded}
+                          disabled={isDelete}
+                          onChange={() =>
+                            toggleTemporaryFlashcardIncluded(
+                              temporaryFlashcard.temporaryFlashcardId
+                            )
+                          }
+                        />
+                      </StyledCheckboxWrapper>
+
+                      <StyledContentContainer
+                        $isTransparent={!temporaryFlashcard.isIncluded}
+                      >
+                        <StyledQuestionHeadline>
+                          Question:
+                        </StyledQuestionHeadline>
+                        <StyledQuestionAnswer>
+                          {" "}
+                          {temporaryFlashcard.question}
+                        </StyledQuestionAnswer>
+                        <StyledAnswerHeadline>Answer:</StyledAnswerHeadline>
+                        <StyledQuestionAnswer>
+                          {temporaryFlashcard.answer}
+                        </StyledQuestionAnswer>
+                      </StyledContentContainer>
+                    </StyledTemporaryFlashcardItem>
+                  );
+                })}
+              </StyledTemporaryFlashcardsList>
+            </>
+          )}
+        </StyledModalContentContainer>
+
+        <StyledConfirmationContainer $morePaddingTop={isDelete}>
+          {!isDelete && temporaryFlashcards.length > 0 && (
+            <ButtonWrapper>
+              <RegularButton
+                type="button"
+                variant="submit"
+                onClick={() => handleSubmitAiFlashcards()}
+              >
+                Save
+              </RegularButton>
+              <RegularButton
+                onClick={toggleDeleteConfirmation}
+                type="button"
+                variant="warning"
+              >
+                Cancel
+              </RegularButton>
+            </ButtonWrapper>
+          )}
+
+          {temporaryFlashcards.length === 0 && (
+            <ButtonWrapper>
+              <RegularButton
+                onClick={cancelFlashcardGeneration}
+                type="button"
+                variant="warning"
+              >
+                Cancel
+              </RegularButton>
+            </ButtonWrapper>
+          )}
+
+          {isDelete && (
+            <DeleteConfirmationDialogContainer>
+              <div>
+                <DeleteConfirmationDialog
+                  onDeleteTemporaryFlashcards={handleDeleteTemporaryFlashcards}
+                  toggleDeleteConfirmation={toggleDeleteConfirmation}
+                  variant="preview collection and all flashcards in it"
+                />
+              </div>
```

Comment:
div not needed

Reply:
I thought I needed this because the elements inside DeleteConfirmationDialog got arranged in a way I didn't want them. But I only had to change the flex-direction of the parent.